src/pip/_internal/req/req_install.py

```diff
@@ -138,6 +139,9 @@ def __init__(
         self.install_options = install_options if install_options else []
         self.global_options = global_options if global_options else []
         self.hash_options = hash_options if hash_options else {}
+        self.ignore_dependencies = ignore_dependencies
+        if isinstance(self.comes_from, InstallRequirement):
+            self.ignore_dependencies = self.comes_from.ignore_dependencies
```

Comment:
Would it be possible for `ignore_dependencies` to be *True*, but `comes_from.ignore_dependencies` is *False*? This current logic would set `self.ignore_dependencies = False` in this case, but I feel it should be *True*?

Reply:
Ok, we can check `comes_from` at first, but how to reproduce this example?

Reply:
self.ignore_dependencies |= self.comes_from.ignore_dependencies

Reply:
with this reqs.txt (aiohttp depends on yarl)

```plain
aiohttp
yarl --no-deps
```
yarl also does not install its dependencies with the previous logic.

Reply:
Don’t use `|=`, that is for bitwise operation, not logical. Use `or` instead.

Or I think this may be easiest:

```python
if isinstance(self.comes_from, InstallRequirement) and self.comes_from.ignore_dependencies:
    self.ignore_dependencies = True
```

Reply:
fixed

Reply:
@uranusjr 
maybe should be rewritten for the consistency according to the implementation above ( like in `get_dependencies`):
```python
 if isinstance(comes_from, InstallRequirement):
    self.ignore_dependencies = self.ignore_dependencies or comes_from.ignore_dependencies
```